examples/post_training/modelopt/mmlu.py

```diff
@@ -151,6 +149,11 @@ def generate_prompt(test_example, dev_examples, few_shots=0, no_subject_prompt=F
     model = get_model(functools.partial(model_provider, parallel_output=True), wrap_with_ddp=False)
     report_current_memory_info()
 
+    # Materialize the model from meta device to gpu before loading the checkpoint.
+    unwrapped_model = unwrap_model(model)[0]
```

Comment:
@kinjalpatel27 was this added to enable weight folding? are these 3 lines necessary?